File: common/src/main/java/cn/breadnicecat/candycraftce/item/CSugarFuels.java
package cn.breadnicecat.candycraftce.item;

import cn.breadnicecat.candycraftce.utils.TickUtils;
import net.minecraft.world.item.ItemStack;
import net.minecraft.world.item.Items;
import org.jetbrains.annotations.NotNull;

import static cn.breadnicecat.candycraftce.item.CDataComponents.SUGAR_BURN_TIME;

public class CSugarFuels {
	
	public static boolean isFuel(@NotNull ItemStack stack) {
		return stack.getItem() == Items.SUGAR || stack.has(SUGAR_BURN_TIME.get());
	}
	
	public static int getBurnDuration(@NotNull ItemStack stack) {
		if (stack.isEmpty()) return 0;
		if (stack.getItem() == Items.SUGAR) return TickUtils.TICK_PER_SEC;
		var v = stack.get(SUGAR_BURN_TIME.get());
		return v != null ? v : 0;
	}
}

File: src/main/kotlin/cn/breadnicecat/candycraftce/core/fuels/CSugarFuels.kt
package cn.breadnicecat.candycraftce.core.fuels

import net.minecraft.world.item.ItemStack

class CSugarFuels {
    companion object {
        fun isFuel(stack: ItemStack): Boolean {
            return getBurnDuration(stack) > 0
        }

        fun getBurnDuration(stack: ItemStack): Int {
            return if (stack.isEmpty) 0 else 20
        }
    }
}